### DISCUSSION TITLE
Error pip install open-webui

### DISCUSSION THREAD

**Discussion by Tamila-2017** [open-webui/open-webui#7496] [General]
Hello all,

I am executing the Open-WebIU installation command from the README.rd:

` pip install open-webui`

but I get the first error:

```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```
Then I follow the recommendation indicated here:

`sudo apt install python3-xyz`

but I get the second error:

```
$ sudo apt install python3-xyz
Reading package lists... Done
Building a dependency tree is ready
Reading information about the composition... Done         
E: Unable to find the python3-xyz
```

That's why I can't install the Open Web UI.
Sorry, but I don't want to use a virtual environment or docker.
I want to use the this installation:
` pip install open-webui`
Please fix it.

**Comment by pierre-josselin** [comment]
The following recommendation:

```
sudo apt install python3-xyz
```

Doesn't mean you have to literally run that command, but that you have to replace xyz with the name of the package in question:

```
sudo apt install python3-open-webui
```

But it won't work anyway because that package doesn't exist.

On the other hand what you can do is to install it with pipx, for example if you are on Debian:

```
sudo apt install pipx
```

Add it to path:

```
pipx ensurepath
```

Then exit and re-open the session.

Then you can install Open WebUI:

```
pipx install open-webui
```

> Please fix it.

Note that this is not a bug at all and the Open WebUI community has nothing to do with this, but the documentation could be improved to take this into account indeed.

**Reply by IvanArkhipov1999** [reply]
Hello, I've followed instructions and got such errors after executing command `pipx install open-webui`:

```
PIP STDOUT
----------

PIP STDERR
----------
ERROR: Could not find a version that satisfies the requirement open-webui (from versions: none)
ERROR: No matching distribution found for open-webui
```

Do you have any ideas how to fix it? I would be grateful for your help!

**Reply by IvanArkhipov1999** [reply]
I built open-webui locally and installed

**Reply by Filiflo** [reply]
I installed python3 with pacman as well as python. The error persists. Maybe a specific python version is needed?

**Reply by Classic298** [reply]
yes python3.11-3.12 as in the docs.

**Reply by lolgamer9743** [reply]
i used `python3.11 -m pip install open-webui` and it works

**Comment by ylcwa** [comment]
same issue here,My OS is Ubuntu-24.04 which maybe not suitable version for open-webui?

**Reply by suresh-senthatti** [reply]
same issue for me also. I'm using open-webui in Ubuntu 24

**Comment by rgaricano** [comment]
pip open-webui require python 3.11,
for install with python 3.12 installed on ubuntu,
- clone & change dir to open-webui root
- build frontend :
```
npm install
NODE_OPTIONS=--max_old_space_size=12000  npm run build
```
- install requirements for backend:

```
cd backend
pip install -r requirements.txt --break-system-packages

```
- run open-webui
`./start.sh`

**Reply by f0rrxst** [reply]
changing my venv to python 3.11 fixed the problem for the. thanks for pointing out the version of python it uses.
